src/controllers/order.controller.js

```diff
@@ -0,0 +1,251 @@
+    } catch (error) {
+      next(error);
+    }
+  }
+
+  /**
+   * Update order (update will call information)
+   * PATCH /orders/:id
+   */
+  async update(req, res, next) {
+    try {
+      const id = parseInt(req.params.id, 10);
+      const { willCallFirstName, willCallLastName } = req.body;
+
+      // Validate required fields
+      if (!willCallFirstName || !willCallLastName) {
+        throw new ValidationError('Invalid request - willCallFirstName and willCallLastName are required');
+      }
+
+      // Check if order exists
+      const order = await this.orderModel.findById(id);
+      if (!order) {
+        throw new NotFoundError('Order not found');
+      }
+
+      // Update reservations associated with this order
+      await this.reservationModel.updateByOrderId(id, {
+        willCallFirstName,
+        willCallLastName,
+      });
+
+      res.status(200).json({ message: 'Order updated' });
+    } catch (error) {
+      next(error);
+    }
+  }
+
+  /**
+   * Process Stripe charge
+   * POST /orders/charge
+   */
+  async processCharge(req, res, next) {
+    try {
+      const { stripeEmail, stripeToken, amount, eventId, metadata } = req.body;
+
+      // Check for blocked email domains
+      if (this._isEmailDomainBlocked(stripeEmail)) {
+        console.log(`Blocked email: ${stripeEmail}`);
+        throw new ValidationError('Unable to process this order.');
+      }
+
+      // Validate pickup party capacity
+      const { pickupPartyId, ticketQuantity } = metadata;
+      const party = await this.pickupPartyModel.findById(parseInt(pickupPartyId, 10));
+
+      if (!party) {
+        throw new NotFoundError('That pickup was not found. Please try again.');
+      }
+
+      const availableCapacity = party.capacity - parseInt(party.reservations, 10);
+
+      if (parseInt(ticketQuantity, 10) > availableCapacity) {
+        throw new ValidationError('Not enough tickets left to complete this order. Please try again or select another pickup location.');
+      }
+
+      // Create Stripe customer
+      const customer = await createCustomer(stripeEmail, stripeToken.id);
+
+      // Create charge
+      const charge = await createCharge({
+        amount,
+        description: eventId,
+        customerId: customer.id,
+        metadata,
+      });
+
+      res.status(200).json(charge);
+    } catch (error) {
+      if (error instanceof StripeError) {
+        // Handle Stripe-specific errors
+        if (error.type === 'StripeCardError') {
+          return res.status(error.statusCode).json({ message: error.message });
+        }
+      }
+      next(error);
+    }
+  }
+
+  /**
+   * Find pickup party by event and location
+   * @private
+   */
+  async _findPickupParty(eventId, pickupLocationId) {
+    const parties = await this.pickupPartyModel.findAll();
+    const party = parties.find(
+      p => p.eventId === eventId && p.pickupLocationId === pickupLocationId
```

Comment:
This seems like it may not scale well. If we fetch 1000 pickupParties to find one its not all that fast of a query. What if we did something like this?
```suggestion
async findByEventAndLocation(eventId, pickupLocationId) {
  const party = await knex(this.table)
    .where({ eventId, pickupLocationId })
    .first();
  return party || null;
}

async _findPickupParty(eventId, pickupLocationId) {
  const party = await this.pickupPartyModel.findByEventAndLocation(eventId, pickupLocationId);
  if (!party) {
    throw new NotFoundError('Pickup party not found for this event and location');
  }
  return party;
}
```